### DISCUSSION TITLE
How to change editorProps.attributes.class?

### DISCUSSION THREAD

**Discussion by appinteractive** [ueberdosis/tiptap#1373] [General Questions] [answered]
How can I change the editorProps.attributes.class? I did not found any way that works to update the inner class attrubte of the editor after initialization. Anyone that had luck with that? The Documentation is not very helpful in that area. @hanspagel

**Comment by philippkuehn** [comment, marked as answer]
Is this not working for you? https://www.tiptap.dev/api/editor/#editor-props

**Reply by appinteractive** [reply]
That's only working on initialization, I need to update It after that. 

My ￼current workaround is to recreate the whole editor but that's very taxing. Especially as I'm unsung multiple instances when toggling their editable state needs to switch there classes.

**Reply by philippkuehn** [reply]
Ah then you probably want to use [`setProps`](https://prosemirror.net/docs/ref/#view.EditorView.setProps) (ProseMirror API).

```js
editor.view.setProps({
  attributes: {
    class: 'newClassName',
  },
})
```
